[docs] Installation improvements (#457)
modm-io/modm: docs/src/guide/installation.md

For Ubuntu 20.04LTS, these commands install the basic build system:

	sudo apt install python3 python3-pip scons cmake git doxygen
	pip3 install modm gdbgui

!!! info "Python Packages in PATH"
	The pip3 command installs executables into `~/.local/bin`, which
	must be added to PATH if it is not already the case.
	Add the following line to the end of your `~/.bashrc` file:

		export PATH="~/.local/bin:$PATH"

For the host system (x68_64/arm64) as a target:

	sudo apt install gcc build-essential libboost-all-dev

For ARM Cortex-M targets:

> 	sudo apt install openocd

Reviewer: Is this OpenOCD release up-to-date?

Reviewer: Seeing the URLs below, should we just add the download and install instructions for OpenOCD too?

Author: > Is this OpenOCD release up-to-date?

Up-to-date for >99% of all modm targets, should work for most people.

Building OpenOCD from Source (or with additional patches) is to complicated for this guide. Maybe we can link to e.g. https://rleh.de/2019/10/08/openocd-stm32-stm32g4-stm32g0.html?

Reviewer: Yeah, good point. Homebrew hides all the dependencies for source compilation, so I forgot about the ugly. Let's link to your blog instead!

!!! warning "Ubuntus 'gcc-arm-none-eabi' package"

> 	The Ubuntu package 'gcc-arm-none-eabi' seems to cause problems at the moment, see https://github.com/modm-io/modm/issues/468.

Reviewer: ```suggestion
	The Ubuntu package 'gcc-arm-none-eabi' causes issues at the moment, see [this Issue](https://github.com/modm-io/modm/issues/468).
```

> 	In the meantime we recommend to use the toolchain provided by ARM as download:

Reviewer: ```suggestion
	In the meantime we recommend using the toolchain provided by ARM:
```

	If you get errors about missing `pyelftools`, check that you're really

> 	running SCons with Python 3!

Reviewer: Is this even necessary at all anymore? Scons 4 works only with Python3, so it won't be choosing the next-best interpreter at runtime: https://scons.org/scons-400-is-available.html

I removed this from the macOS guide for this reason.

Author: But Ubuntu provides [Scons 3](https://packages.ubuntu.com/focal/scons).

Ubuntu 20.04 has Python3 and therefore no problem anyway.
I would keep the warning for now to give a hint to users of Ubuntu ≤ 19.10.

Reviewer: Ok, good point. I'll squash and merge then.
